File: Lecture003 If-Else, While loop & Lots of Patterns/Pattern18.java
package com.company;

import java.util.Scanner;

public class Pattern18 {
    // Pattern 18
    public static void main(String [] args){
        /*
        Pattern 18
                                  **********
                                  ****  ****
                                  ***    ***
                                  **      **
                                  *        *
                                  *        *
                                  **      **
                                  ***    ***
                                  ****  ****
                                  **********
         */
        // Creating Scanner Object For taking inputs from the USER
        Scanner sc = new Scanner(System.in);
        // Take Number of ROWS and COLUMNS from the USER
        System.out.print(" Enter Number of ROWS for the Pattern: ");
        int n = sc.nextInt();
        pattern(n);
    }
    static void pattern(int n){
        // Procedure to print the Required Pattern
        int inSpaces = 2;
        int belowSpaces = 2 * n - 2;
        for(int row =1; row <= 2 * n ; row++){

            int totalColumns = row > n ? row - n: n - row + 1;
            for(int column = 1; column <= totalColumns; column++){
                System.out.print("*");
            }
            if(row <= n && row != 1) {
                for (int spaces = 1; spaces <= inSpaces; spaces++) {
                    System.out.print(" ");
                }
                inSpaces = inSpaces + 2;
            }
            if(row > n && row != 2 * n){
                for (int spaces = 1; spaces <= belowSpaces; spaces++) {
                    System.out.print(" ");
                }
                belowSpaces = belowSpaces - 2;
            }


            for(int column = 1; column <= totalColumns; column++){
                System.out.print("*");
            }

            System.out.println();
        }
    }
}

File: 003_If-Else_While_loop_and_Lots_of_Patterns/Swift/patterns/Pattern18.swift

import Foundation

// Function to print the specified pattern
func printPattern(_ n: Int) {
    var inSpaces = 2
    var belowSpaces = 2 * n - 2
    for row in 1...(2 * n) {
        let totalColumns = row > n ? row - n : n - row + 1
        for _ in 1...totalColumns {
            print("*", terminator: "")
        }
        if row <= n && row != 1 {
            for _ in 1...inSpaces {
                print(" ", terminator: "")
            }
            inSpaces += 2
        }
        if row > n && row != 2 * n {
            for _ in 1...belowSpaces {
                print(" ", terminator: "")
            }
            belowSpaces -= 2
        }
        for _ in 1...totalColumns {
            print("*", terminator: "")
        }
        print("")
    }
}

// Taking user input for the number of rows
print("Enter Number of ROWS for the Pattern: ", terminator: "")
if let input = readLine(), let n = Int(input) {
    printPattern(n)
} else {
    print("Invalid input")
}
